app/code/Mykhailok/SupportChat/view/frontend/web/js/action/save-message.js

```diff
@@ -0,0 +1,39 @@
+define([
+    'jquery'
+], function ($) {
+    'use strict';
+
+    /**
+     * @param {Object} data
+     * @param {String} url
+     */
+    return function (data, url, self) {
```

Comment:
@mykhailok-dv Do not pass objects to other objects if possible. Pass callbacks or use the returned value to add callbacks. Check what jQuery.ajax returns. It is possible to add callbacks in the view component in the same way as `done` callback was added - https://api.jquery.com/jQuery.ajax/#jqXHR
By the way, using `done` in the lesson was incorrect - it should be `always`.

Reply:
Was been pass methods .done().fail() after function saveMessage() returned jQuery.ajax.